AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1528,26 +1530,11 @@ - (void)measureNodeWithBoundsIfNecessary:(CGRect)bounds
   
     // Use as default constrained size the bounds
     CGSize parentSize = bounds.size;
-    ASSizeRange constrainedSize = ASSizeRangeMake(parentSize);
-    
-    // Checkout if constrained size of pending or calculated display node layout can be used
-    if (_pendingDisplayNodeLayout != nullptr && CGSizeEqualToSize(_pendingDisplayNodeLayout->layout.size, bounds.size)) {
-      // We assume the size from the last returned layoutThatFits: layout was applied so use the pejnding display node
-      // layout constrained size
-      constrainedSize = _pendingDisplayNodeLayout->constrainedSize;
-    } else if (CGSizeEqualToSize(_calculatedDisplayNodeLayout->layout.size, CGSizeZero) == NO &&
-               CGSizeEqualToSize(_calculatedDisplayNodeLayout->layout.size, bounds.size)) {
-      // We assume the  _calculatedDisplayNodeLayout is still valid and the frame is not different
-      constrainedSize = _calculatedDisplayNodeLayout->constrainedSize;
-    } else {
-      // In this case neither the _pendingDisplayNodeLayout or the _calculatedDisplayNodeLayout constrained size can
-      // be reused, so the current bounds is used. This is usual the case if a frame was set manually that differs to
-      // the one returned from layoutThatFits: or layoutThatFits: was never called
-    }
+    ASSizeRange constrainedSize = [self constrainedSizeForLayoutPass];
     
     return std::make_shared<ASDisplayNodeLayout>(
       [self calculateLayoutThatFits:constrainedSize restrictedToSize:self.style.size relativeToParentSize:parentSize],
-      constrainedSize,
+      [self constrainedSizeForLayoutPass],
```

Comment:
@maicki this one should just use the local variable, not call the method again, right?